consensus/state_processing/src/common/get_attestation_participation.rs

```diff
@@ -0,0 +1,46 @@
+use crate::per_block_processing::errors::BlockProcessingError as Error;
+use integer_sqrt::IntegerSquareRoot;
+use safe_arith::SafeArith;
+use smallvec::SmallVec;
+use types::consts::altair::{
+    NUM_FLAG_INDICES, TIMELY_HEAD_FLAG_INDEX, TIMELY_SOURCE_FLAG_INDEX, TIMELY_TARGET_FLAG_INDEX,
+};
+use types::{AttestationData, BeaconState, ChainSpec, EthSpec};
+
+/// Get the participation flags for a **valid** attestation.
+///
+/// You must have called `verify_attestation_for_block_inclusion` or similar before
+/// calling this function, in order to ensure that the attestation's source is correct.
+///
+/// This function is extracted from `process_attestation`
+pub fn get_attestation_participation<T: EthSpec>(
+    data: &AttestationData,
+    state: &BeaconState<T>,
+    spec: &ChainSpec,
+) -> Result<SmallVec<[u32; NUM_FLAG_INDICES]>, Error> {
+    // Matching roots.
+    // Source match is checked by `verify_attestation_for_block_inclusion`.
+    let is_matching_head = data.beacon_block_root == *state.get_block_root(data.slot)?;
+    let is_matching_source = true;
+    let is_matching_target =
+        data.target.root == *state.get_block_root_at_epoch(data.target.epoch)?;
+
+    // Participation flag indices
+    let mut participation_flag_indices = SmallVec::new();
+    if is_matching_head
+        && is_matching_target
+        && state.slot() <= data.slot.safe_add(spec.min_attestation_inclusion_delay)?
```

Comment:
```suggestion
        && state.slot() == data.slot.safe_add(spec.min_attestation_inclusion_delay)?
```

Fixes a slight difference from the [spec](https://github.com/ethereum/eth2.0-specs/blob/v1.1.0-alpha.3/specs/altair/beacon-chain.md#modified-process_attestation).

I don't think my suggested change would *actually* make a change at runtime, since attestations from prior to the inclusion distance will be filtered out before hitting this function. That goes for both `per_block_processing` and `operation_pool.`

Reply:
Resolved in https://github.com/sigp/lighthouse/pull/2279/commits/6415962290a54c00289cb526aefc5a304e9e94e4